### PR TITLE
Custom file upload MIME type

### LINKED ISSUE #668: No way to set content type of each param in multipart
Request Method: POST
Referrer Policy: no-referrer-when-downgrade
Cache-Control: no-cache, no-store, must-revalidate
Connection: keep-alive
Content-Length: 579
Content-Type: application/json
Date: Mon, 02 Apr 2018 09:47:17 GMT
Server: Apigee Router
Accept: application/json
Accept-Encoding: gzip, deflate
Accept-Language: en-US,en;q=0.9
Connection: keep-alive
Content-Length: 478592
Content-Type: multipart/form-data; boundary=----WebKitFormBoundaryV2XWWBaHohtZIDEg
User-Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/65.0.3325.181 Safari/537.36
X-Requested-With: XMLHttpRequest
------WebKitFormBoundaryV2XWWBaHohtZIDEg
Content-Disposition: form-data; name="guid"

9ffffba4-2f14-4e93-9c59-c03e67262b47
------WebKitFormBoundaryV2XWWBaHohtZIDEg
Content-Disposition: form-data; name="json"
// Ability to set the content type for each part

{"name":"test.png"}
------WebKitFormBoundaryV2XWWBaHohtZIDEg
Content-Disposition: form-data; name="qquuid"

d07f56d9-5b41-49f9-8fd1-d1ea29be4a16
------WebKitFormBoundaryV2XWWBaHohtZIDEg
Content-Disposition: form-data; name="qqfilename"

test.png
------WebKitFormBoundaryV2XWWBaHohtZIDEg
Content-Disposition: form-data; name="qqtotalfilesize"

477560
------WebKitFormBoundaryV2XWWBaHohtZIDEg
Content-Disposition: form-data; name="file"; filename="test.png"
Content-Type: image/png

------WebKitFormBoundaryV2XWWBaHohtZIDEg--

### PR DESCRIPTION
**Pull request by skuro**
Fixes #668 

In order to specify a custom MIME type per each uploaded file, I've introduced a syntax that mimicks what's [already possible in cURL](https://curl.haxx.se/docs/manpage.html#-F):

```bash
$ http --form POST httpbin.org/post file1@'/path/to/file.bin;type=application/zip' file2@'/path/to/file2.bin;type=image/png'
```

### CODE REVIEW

File: httpie/cli/constants.py [hunk 1]
```diff
@@ -24,6 +24,7 @@
 SEPARATOR_DATA_STRING = '='
 SEPARATOR_DATA_RAW_JSON = ':='
 SEPARATOR_FILE_UPLOAD = '@'
+SEPARATOR_FILE_UPLOAD_TYPE = ';type='
```
Comment (jakubroztocil) [inline httpie/cli/constants.py:27]:
```suggestion
SEPARATOR_FILE_UPLOAD_TYPE = ';type='  # in already parsed file upload path only

File: tests/test_uploads.py [hunk 2]
```diff
@@ -38,6 +38,16 @@ def test_upload_multiple_fields_with_the_same_name(self, httpbin):
         assert r.count(FILE_CONTENT) in [3, 4]
         assert r.count('Content-Type: text/plain') == 2
 
+    def test_upload_custom_content_type(self, httpbin):
+        r = http('--form', '--verbose', 'POST', httpbin.url + '/post',
+                 'test-file@%s;type=image/vnd.microsoft.icon' % FILE_PATH_ARG)
+        assert HTTP_OK in r
+        # Content type is stripped from the filename
+        assert 'Content-Disposition: form-data; name="test-file";' \
+               ' filename="%s"' % os.path.basename(FILE_PATH) in r
```
Comment (jakubroztocil) [inline tests/test_uploads.py:47]: You can use f-strings!

**Comment by jakubroztocil** [comment]
One more thing — see `http --help`. There we have:

```
      '@' Form file fields (only with --form, -f):

          cs@~/Documents/CV.pdf
```


So the new feature should also be mentioned there:


```
      '@' Form file fields (only with --form, -f):

          cv@~/Documents/CV.pdf
          cv@'~/Documents/CV.pdf;type=application/pdf'
```

File: tests/test_uploads.py (hunk 2 above)
Comment (skuro) [reply tests/test_uploads.py:47]: You can tell Python is not my main language, can you? :o)
Will fix

Comment (jakubroztocil) [reply tests/test_uploads.py:47]: Haha. But this is the only thing that gave you away :)

**Comment by skuro** [comment]
Alright, I think I have incorporated all feedback from @jakubroztocil. Let me know if it Looks Good(tm) now or if there's anything else I can add

**Comment by jakubroztocil** [comment]
All good, thanks! 🍰

**Pull request merged**